R/pkg/R/DataFrame.R

```diff
@@ -1848,3 +1848,28 @@ setMethod("crosstab",
             sct <- callJMethod(statFunctions, "crosstab", col1, col2)
             collect(dataFrame(sct))
           })
+
+
+#' This function downloads the contents of a DataFrame into an R's data.frame.
+#' Since data.frames are held in memory, ensure that you have enough memory
+#' in your system to accommodate the contents.
+#' 
+#' @title Download data from a DataFrame into a data.frame
+#' @param x a DataFrame
+#' @return a data.frame
+#' @rdname as.data.frame
+#' @examples \dontrun{
+#' 
+#' irisDF <- createDataFrame(sqlContext, iris)
+#' df <- as.data.frame(irisDF[irisDF$Species == "setosa", ])
+#' }
+setGeneric("as.data.frame")
+setMethod(f = "as.data.frame", signature = "DataFrame", definition =
```

Comment:
Minor style comments -- To keep this similar to the other methods in this function I think this can be 
```
setMethod("as.data.frame",
                   signature(x = "DataFrame"),
                   function(x, ...) {
}
```